internal/datastore/memdb/memdb.go

```diff
@@ -108,8 +112,13 @@ func (mdb *memdbDatastore) SnapshotReader(revision datastore.Revision) datastore
 		revIndex = len(mdb.revisions) - 1
 	}
 
-	snapshotRevision := mdb.revisions[revIndex].revision
-	roTxn := mdb.revisions[revIndex].db.Txn(false)
+	rev := mdb.revisions[revIndex]
+	if rev.db == nil {
```

Comment:
an alternative would be to have a `closed` bool on the datastore, which would could check at the top instead (since this and `.Close()` are both protected by the mutex)

no strong preference though, if you prefer this method.

Reply:
I'd likely still want the `nil` check just to be on the safe side, but I can add a `closed` bool if you like